ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/execution/VariableExecutorTest.java

```diff
@@ -0,0 +1,168 @@
+        engine.configure(sql),
+        sessionProperties,
+        engine.getEngine(),
+        engine.getServiceContext()
+    );
+    assertThat(response, is(Optional.empty()));
+  }
+
+  private Optional<KsqlEntity> executeUndefineVariable(final String sql) {
+    return CustomExecutors.UNDEFINE_VARIABLE.execute(
+        engine.configure(sql),
+        sessionProperties,
+        engine.getEngine(),
+        engine.getServiceContext()
+    );
+  }
+
+  @Test
+  public void shouldSetVariables() {
+    // When:
+    executeDefineVariable("DEFINE var1 = 'John Peter';");
+    executeDefineVariable("DEFINE var2 = '''John Peter''';");
+
+    // Then:
+    final Map<String, String> variablesMap = sessionProperties.getSessionVariables();
+    assertThat(variablesMap.size(), is(2));
+    assertThat(variablesMap, hasEntry("var1", "John Peter"));
+    assertThat(variablesMap, hasEntry("var2", "'John Peter'"));
+  }
+
+  @Test
+  public void shouldSetCaseInsensitiveVariables() {
+    // When:
+    executeDefineVariable("DEFINE A = 'val1';");
+    executeDefineVariable("DEFINE b = 'val2';");
+
+    // Then:
+    final Map<String, String> variablesMap = sessionProperties.getSessionVariables();
+    assertThat(variablesMap.containsKey("a"), is(true));
+    assertThat(variablesMap.get("a"), is("val1"));
+    assertThat(variablesMap.containsKey("A"), is(true));
+    assertThat(variablesMap.get("A"), is("val1"));
+    assertThat(variablesMap.containsKey("b"), is(true));
+    assertThat(variablesMap.get("b"), is("val2"));
+    assertThat(variablesMap.containsKey("B"), is(true));
+    assertThat(variablesMap.get("B"), is("val2"));
+  }
+
+  @Test
+  public void shouldUnsetVariables() {
+    // Given:
+    sessionProperties.setVariable("var1", "1");
+    sessionProperties.setVariable("var2", "2");
+
+    // When:
+    final Optional<KsqlEntity> response = executeUndefineVariable("UNDEFINE var1;");
+    assertThat(response, is(Optional.empty()));
+
+    // Then:
+    final Map<String, String> variablesMap = sessionProperties.getSessionVariables();
+    assertThat(variablesMap.size(), is(1));
+    assertThat(variablesMap, hasEntry("var2", "2"));
+  }
+
+  @Test
+  public void shouldUnsetCaseInsensitiveVariables() {
+    // Given:
+    sessionProperties.setVariable("VAR1", "1");
+
+    // When:
+    final Optional<KsqlEntity> response = executeUndefineVariable("UNDEFINE var1;");
+    assertThat(response, is(Optional.empty()));
+
+    // Then:
+    final Map<String, String> variablesMap = sessionProperties.getSessionVariables();
+    assertThat(variablesMap.size(), is(0));
+  }
+
+  @Test
+  public void shouldReturnWarningWhenUndefineAnUnknownVariable() {
+    // When:
+    final KsqlEntity response = executeUndefineVariable("UNDEFINE var1;").get();
+
+    // Then:
+    assertThat(((WarningEntity)response).getMessage(),
+        containsString("Cannot undefine variable 'var1' which was never defined"));
+  }
+
+  @Test
+  public void shouldThrowOnInvalidValues() {
```

Comment:
should we add a test on adding duplicates? we should also check to make sure that case insensitive duplicates are stilled rejected

Reply:
Variables will be overridden (no rejected) if an existing case insensitive variable exists. I'll add a test case that verifies that. 

It will allow us to do this:
```
DEFINE var1 = '1';
DEFINE VAR1 = '2';
DEFINE vAr1 = '3'; // latest update
```